src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1166,6 +1171,31 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
             return expression;
         }
 
+        /// <summary>
+        /// Is this is an _ identifier in a context where discards are allowed?
+        /// </summary>
+        private bool FallBackOnDiscard(SimpleNameSyntax node, DiagnosticBag diagnostics)
+        {
+            if (node.Identifier.ContextualKind() != SyntaxKind.UnderscoreToken)
+            {
+                return false;
+            }
+
+            CSharpSyntaxNode containingDeconstruction = node.GetContainingDeconstruction();
+            return containingDeconstruction != null || IsOutVarDiscardIdentifier(node);
+        }
+
+        private bool IsOutVarDiscardIdentifier(SimpleNameSyntax node)
+        {
+            Debug.Assert(node.Identifier.ContextualKind() == SyntaxKind.UnderscoreToken);
+
+            CSharpSyntaxNode parent = node.Parent;
+            return parent.Kind() == SyntaxKind.Argument &&
+                ((ArgumentSyntax)parent)?.RefOrOutKeyword.Kind() == SyntaxKind.OutKeyword &&
+                (parent = parent.Parent)?.Kind() == SyntaxKind.ArgumentList &&
+                (parent = parent.Parent)?.Kind() == SyntaxKind.InvocationExpression;
```

Comment:
Yes, I did not think of those. I'll add tests. Thanks!

---
In reply to: [88535260](https://github.com/dotnet/roslyn/pull/15192#discussion_r88535260) [](ancestors = 88535260)